examples/blake2f.rs

```diff
@@ -0,0 +1,1459 @@
+            .collect();
+
+        let final_inputs = FinalInput {
+            round: F::from(values.round as u64),
+            v_vec: v_vec_values.iter().map(|&v| F::from(v)).collect(),
+            h_vec: h_vec_values.iter().map(|&v| F::from(v)).collect(),
+            output_vec: output_vec_values.iter().map(|&v| F::from(v)).collect(),
+            v_split_bit_vec: v_vec_values
+                .iter()
+                .map(|&v| split_value_4bits(v as u128, BITS_NUMBER))
+                .collect(),
+            h_split_bit_vec: h_vec_values
+                .iter()
+                .map(|&v| split_value_4bits(v as u128, BITS_NUMBER))
+                .collect(),
+            v_xor_split_bit_vec: v_vec_values[0..V_LEN as usize / 2]
+                .iter()
+                .zip(v_vec_values[V_LEN as usize / 2..V_LEN as usize].iter())
+                .map(|(&v1, &v2)| split_xor_value(v1, v2))
+                .collect(),
+            final_split_bit_vec: output_vec_values
+                .iter()
+                .map(|&output| split_value_4bits(output as u128, BITS_NUMBER))
+                .collect(),
+        };
+        ctx.add(&blake2f_final_step, final_inputs);
+        // ba80a53f981c4d0d, 6a2797b69f12f6e9, 4c212f14685ac4b7, 4b12bb6fdbffa2d1
+        // 7d87c5392aab792d, c252d5de4533cc95, 18d38aa8dbf1925a,b92386edd4009923
+        println!("output = {:x?}", output_vec_values);
+    })
+}
+
+fn blake2f_super_circuit<F: PrimeField + Hash>() -> SuperCircuit<F, InputValues> {
+    super_circuit::<F, InputValues, _>("blake2f", |ctx| {
+        let single_config = config(SingleRowCellManager {}, SimpleStepSelectorBuilder {});
+        let (_, iv_table) =
+            ctx.sub_circuit(single_config.clone(), blake2f_iv_table, IV_LENGTH as usize);
+        let (_, bits_table) = ctx.sub_circuit(
+            single_config.clone(),
+            blake2f_4bits_table,
+            BITS_NUMBER as usize,
+        );
+        let (_, xor_4bits_table) = ctx.sub_circuit(
+            single_config,
+            blake2f_xor_4bits_table,
+            (BITS_NUMBER * BITS_NUMBER) as usize,
+        );
+
+        let maxwidth_config = config(
+            MaxWidthCellManager::new(250, true),
+            SimpleStepSelectorBuilder {},
+        );
+
+        let params = CircuitParams {
+            iv_table,
+            bits_table,
+            xor_4bits_table,
+        };
+        let (blake2f, _) = ctx.sub_circuit(maxwidth_config, blake2f_circuit, params);
+
+        ctx.mapping(move |ctx, values| {
+            ctx.map(&blake2f, values);
+        })
+    })
+}
+
+fn main() {
+    let super_circuit = blake2f_super_circuit::<Fr>();
+    let compiled = chiquitoSuperCircuit2Halo2(&super_circuit);
+
+    let values = InputValues {
+        round: 12,
+        // h[0] = hex"48c9bdf267e6096a 3ba7ca8485ae67bb 2bf894fe72f36e3c f1361d5f3af54fa5";
+        // h[1] = hex"d182e6ad7f520e51 1f6c3e2b8c68059b 6bbd41fbabd9831f 79217e1319cde05b";
+        h_vec: [
```

Comment:
@10to4 the values in h_vec does not seem to match the comment and the EIP, why is that?

Reply:
Is it because of the endianness?

Reply:
Yes, I can continue to change the output order.

Reply:
Ok